ports/espressif/common-hal/_bleio/Characteristic.c

```diff
@@ -118,10 +206,32 @@ bleio_characteristic_properties_t common_hal_bleio_characteristic_get_properties
     return self->props;
 }
 
-void common_hal_bleio_characteristic_add_descriptor(bleio_characteristic_obj_t *self, bleio_descriptor_obj_t *descriptor) {
+void common_hal_bleio_characteristic_add_descriptor(bleio_characteristic_obj_t *self,
+    bleio_descriptor_obj_t *descriptor) {
     // TODO: Implement this.
+
+    mp_obj_list_append(MP_OBJ_FROM_PTR(self->descriptor_list),
+        MP_OBJ_FROM_PTR(descriptor));
 }
```

Comment:
Is there more to implement here?

Reply:
Yes, this is where code to add a descriptor to a characteristic for a GATT server goes. So, it should be fine as-is for GATT client.